From 9c36df7d2269be8428249202fbef57f614b01a34 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:08:26 -0400
Subject: [usb] sierra: do not resume I/O on closed ports

Message-id: <1410887463-20674-435-git-send-email-dzickus@redhat.com>
Patchwork-id: 95265
O-Subject: [RHEL7 PATCH 434/591] USB: sierra: do not resume I/O on closed ports
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit a283d080a49a5014f525bf722e5a6a07835e45ef
Author: Johan Hovold <jhovold@gmail.com>
Date:   Mon May 26 19:23:03 2014 +0200

    USB: sierra: do not resume I/O on closed ports

    Do not resume any I/O, including the delayed write queue, on closed
    ports.

    Note that this currently has no functional impact due to the
    usb_autopm_get_interface() in close(), but that call is about to be
    removed by a follow-up patch.

    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/sierra.c b/drivers/usb/serial/sierra.c
index 74b417c..ac5e20d 100644
--- a/drivers/usb/serial/sierra.c
+++ b/drivers/usb/serial/sierra.c
@@ -1013,7 +1013,7 @@ static int sierra_resume(struct usb_serial *serial)
   port = serial->port[i];
   portdata = usb_get_serial_port_data(port);
 
-  if (!portdata)
+  if (!portdata || !portdata->opened)
    continue;
 
   while ((urb = usb_get_from_anchor(&portdata->delayed))) {
@@ -1036,11 +1036,9 @@ static int sierra_resume(struct usb_serial *serial)
    }
   }
 
-  if (portdata->opened) {
-   err = sierra_submit_rx_urbs(port, GFP_ATOMIC);
-   if (err)
-    ec++;
-  }
+  err = sierra_submit_rx_urbs(port, GFP_ATOMIC);
+  if (err)
+   ec++;
  }
  intfdata->suspended = 0;
  spin_unlock_irq(&intfdata->susp_lock);
-- 
1.7.1